windowsforms/FlexGridShowcaseDemo/cs/MainForm.cs

```diff
@@ -0,0 +1,682 @@
+            _ribbonMenuFormatting.IconSet.Add(image);
+
+            // Columns
+            image = new C1.Framework.C1BitmapIcon("Columns", new Size(20, 20), Color.Transparent, LoadImage("Columns"));
+            _ribbonMenuColumns.IconSet.Add(image);
+
+            // Filter
+            image = new C1.Framework.C1BitmapIcon("Filter", new Size(20, 20), Color.Transparent, LoadImage("Filter"));
+            _ribbonButtonFilter.IconSet.Add(image);
+
+            var appIcon = Properties.Resources.App;
+            Icon = appIcon;
+        }
+        private void InitGroups()
+        {
+            var propertyNames = new List<string>();
+            if (_ribbonCheckBoxGroupByCountry.Checked)
+            {
+                propertyNames.Add("Country");
+            }
+            if (_ribbonCheckBoxGroupByProduct.Checked)
+            {
+                propertyNames.Add("Product");
+            }
+
+            // Clear condition filters
+            if (_flexGrid.GroupDescriptions != null && propertyNames.Count == 0)
+            {
+                _flexGrid.GroupDescriptions = null;
+                return;
+            }
+
+            var groups = propertyNames
+                    .Select(x => new GroupDescription(x, ListSortDirection.Ascending))
+                    .ToList();
+            _flexGrid.GroupDescriptions = groups;
+        }
+
+        private void InitFlexGrid()
+        {
+            // setup flexgrid
+            _flexGrid.AllowFiltering = true;
+            _flexGrid.AllowFiltering = true;
+            _flexGrid.AllowMerging = AllowMergingEnum.Nodes;
+            _flexGrid.HideGroupedColumns = true;
+            _flexGrid.ShowErrors = true;
+
+            var columns = _flexGrid.Cols;
+
+            // setup flexgrid columns
+            columns[0].Width = 22;
+
+            var idColumn = columns["ID"];
+            idColumn.Width = 50;
+            idColumn.AllowEditing = false;
+
+            // setup combo list
+            var products = (from s in _dataSet.Tables["Products"].Rows.Cast<DataRow>() select s)
+                .Select(x => x["Name"].ToString());
+            columns["Product"].ComboList = string.Join("|", products);
+
+            // build image map for countries
+            var flagImageMap = new Dictionary<Countries, Image>();
+            foreach (Countries country in Enum.GetValues(typeof(Countries)))
+            {
+                flagImageMap.Add(country, LoadImage(country.ToString()));
+            }
+
+            var countryColumn = columns["Country"];
+
+            // assign image map to country column
+            countryColumn.ImageMap = flagImageMap;
+            countryColumn.ImageAndText = true;
+
+            // build image map for colors
+            var colorImageMap = new Dictionary<DrawColors, Image>();
+            foreach (DrawColors color in Enum.GetValues(typeof(DrawColors)))
```

Comment:
Rename the enum to be singular.

Reply:
Fixed with f005546a5824888644a64d6c3746cb164529c9dd.